### PR TITLE
#1707 - Application Form: Additional Transportation Costs (FT & PT)

### PR DESCRIPTION
**Pull request by sh16011993**
**As a part of this PR, formio definitions were updated to include the additional transport cost related questions for all program years.**

**Screenshot:**

image

Please note: There is no field validation currently in place for the 1st question (as per the story requirements) - "Please estimate your weekly transportation costs."

### CODE REVIEW

File: sources/packages/forms/src/form-definitions/sfaa2021-22.json [hunk 1]
```diff
@@ -31148,6 +31148,154 @@
...
+                  "decimalLimit": 2,
+                  "requireDecimal": true,
+                  "validate": {
+                    "required": true,
+                    "multiple": "",
+                    "custom": ""
+                  },
+                  "conditional": {
+                    "show": "true",
+                    "when": "strongYouMayBeEligibleForAnAdditionalTransportationAllowanceIfThereIsNoLimitedLocalPublicTransitWhereYouLiveOrIfYouAreRequiredDueToSpecialCircumstancesDoYouWantToBeAssessedForAdditionalTransportationCostsStrong",
```
Comment (andrepestana-aot) [inline sources/packages/forms/src/form-definitions/sfaa2021-22.json:31177]: Please change this long key to something shorter and check the impacted points (I don't think there will be great impacts).

File: sources/packages/forms/src/form-definitions/sfaa2021-22.json [hunk 2]
```diff
@@ -31148,6 +31148,154 @@
...
+                  "properties": {},
+                  "lockKey": true
+                },
+                {
+                  "autofocus": false,
+                  "input": true,
+                  "tableView": false,
+                  "inputType": "number",
+                  "label": "Please estimate your weekly travel distance in kilometers:",
+                  "key": "studentDataAdditionalTransportKm",
```
Comment (andrewsignori-aot) [inline sources/packages/forms/src/form-definitions/sfaa2021-22.json:31192]: The `studentData` prefix is added while loading the values on Camunda.
For form.io API keys, we can use the variable names provided in the ticket without the prefix.
- additionalTransportKm
- additionalTransportCost
- additionalTransportWeeks
- additionalTransportPlacement

**Review by andrewsignori-aot — COMMENTED** [review]
Nice work, please take a look at the comments.

File: sources/packages/load-test/src/_tests_/form-io/form-io-application-submission/models.ts [hunk 3]
```diff
@@ -80,8 +80,7 @@ export const APPLICATION_SUBMISSION_PAYLOAD = {
     livingWithParents: "no",
     strongIfYouMustRelocateToADifferentCityToAttendSchoolDoYouWantToProvideACostToBeConsideredStrong:
       "no",
-    strongYouMayBeEligibleForAnAdditionalTransportationAllowanceIfThereIsNoLimitedLocalPublicTransitWhereYouLiveOrIfYouAreRequiredDueToSpecialCircumstancesDoYouWantToBeAssessedForAdditionalTransportationCostsStrong:
-      "no",
+    eligibleForAnAdditionalTransportationAllowance: "no",
```
Comment (andrewsignori-aot) [inline sources/packages/load-test/src/_tests_/form-io/form-io-application-submission/models.ts:83]: Thanks for taking care of this ❤️

**Review by andrewsignori-aot — APPROVED** [review]
Thanks for doing the changes, looks good 👍

**Review by andrepestana-aot — APPROVED** [review]
Looks good! Thanks for taking care of the var names.

**Review by guru-aot — APPROVED** [review]
LGTM, nice work @sh16011993

**Review by lewischen-aot — APPROVED** [review]
LGTM . Nice work 👍

**Review by dheepak-aot — APPROVED** [review]
Looks good. Nice work @sh16011993

**Pull request merged**